core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -2888,6 +2888,19 @@ class GroupCoordinatorTest {
     assertEquals(Errors.NONE, thirdResult.error)
   }
 
+  @Test
+  def testCompleteHeartbeatWithGroupDead(): Unit = {
```

Comment:
Could we also have a test case for the scenario where tryComplete is called for the heartbeat of a member that was already removed from the group?